Fix tour breaking when using manipulating comparison mode b-state

ZachTRice opened the pull request:
## Description

Fixes #1685

- If mode has switched to B state by the user, switch back to a state before parsing the tour step.
- Add some validation if there are no layers in the layer list (additional bug breaking in current production)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/master/.github/CONTRIBUTING.md) doc
- [x] I have added necessary documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] Any dependent changes have been merged and published in downstream modules (if applicable)

## Further comments

If this is a relatively large or complex change, start a discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

@nasa-gibs/worldview

Benjaki2 commented on web/js/map/ui.js:
```diff
@@ -565,7 +565,8 @@ export function mapui(models, config) {
     });
-    if (!layer && layers[0].get('group')) {
+
+    if (!layer && layers.length && layers[0].get('group')) {
```
Might want a `layers.length > 0` as well to ensure layers[0] exists

ZachTRice replied:
`layers.length` and `layers.length > 0` are the same. The former being slightly faster.
ref: https://stackoverflow.com/questions/32911424/array-length-vs-array-length-0

Benjaki2 replied:
Whoops. Did not know. 👍

ZachTRice replied:
All good 😉 let me try rebasing on the new tour story and see if I can reproduce

ZachTRice commented:
Hey @Benjaki2 I just realized the reason you are seeing an error with the new tour is because I added a fix in Minnie's branch. I just added that fix over in this branch so if you check this PR again with the new tour, you should not see it breaking. 

Really good catch!
